vector/src/main/java/im/vector/app/core/event/GetTimelineEventUseCase.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.event
+
+import androidx.lifecycle.asFlow
+import im.vector.app.core.di.ActiveSessionHolder
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import org.matrix.android.sdk.api.session.room.timeline.TimelineEvent
+import org.matrix.android.sdk.flow.unwrap
+import javax.inject.Inject
+
+// TODO add unit tests
+class GetTimelineEventUseCase @Inject constructor(
+        private val activeSessionHolder: ActiveSessionHolder,
+) {
+
+    fun execute(roomId: String, eventId: String): Flow<TimelineEvent> {
+        return activeSessionHolder
+                .getActiveSession()
+                .roomService()
+                .getRoom(roomId)
+                ?.timelineService()
+                ?.getTimelineEventLive(eventId)
+                ?.asFlow()
+                ?.unwrap()
+                ?: emptyFlow()
```

Comment:
The indentation is a bit weird, I think we can reduce the lines count like this
```suggestion
        return activeSessionHolder.getActiveSession().roomService().getRoom(roomId)
                ?.timelineService()
                ?.getTimelineEventLive(eventId)
                ?.asFlow()
                ?.unwrap()
                ?: emptyFlow()
```

Reply:
Yes updated in [055ed35](https://github.com/vector-im/element-android/pull/8069/commits/055ed35fa6567b617053d57096cc5a86d8b34476)